### PR TITLE
[9.0] Add sale_datetime_confirm

### PR DESCRIPTION
**Pull request by florian-dacosta**
Since version 9, the date_confirm field has been removed from sale orders.
This fields seems important to me since you can confirm a sale order many days after receiving it (If you wait for it to be paid by transfer, check or for any other reason.)
It is nice to know exactly when the sale order was confirmed anyway.

The module also compute the planned date of procurements based on the sale order confirm date.

### CODE REVIEW

File: sale_datetime_confirm/README.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+=====
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/167/9.0
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/
```
Comment (elicoidal) [inline sale_datetime_confirm/README.rst]: please update to newest template

File: sale_datetime_confirm/README.rst [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+Contributors
+------------
+
+* Florian da Costa <florian.dacosta@akretion.com>
+
+Maintainer
+----------
+
+.. image:: http://odoo-community.org/logo.png
```
Comment (elicoidal) [inline sale_datetime_confirm/README.rst]: same: use newest template

File: sale_datetime_confirm/__init__.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# This program is distributed in the hope that it will be useful,       #
+# but WITHOUT ANY WARRANTY; without even the implied warranty of        #
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         #
+# GNU Affero General Public Licensefor more details.                    #
+#                                                                       #
+# You should have received a copy of the                                #
+# GNU Affero General Public License                                     #
+# along with this program.  If not, see <http://www.gnu.org/licenses/>. #
+#                                                                       #
+#########################################################################
```
Comment (elicoidal) [inline sale_datetime_confirm/__init__.py]: @florian-dacosta   Thanks for your contribution to the OCA. 
Let me introduce to you some basic information about contribution in OCA and I will review other points in more comments inside the code.
In general, please follow the principles: 
* Contribution guidelines: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md
* Please follow the OCA README template:
https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst
* We recommend to include a screenshot as well to help new comers (nice to have)
* Please use simple headers in all py files following OCA templates
https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

File: sale_datetime_confirm/README.rst [hunk 4]
```diff
@@ -0,0 +1,49 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=====================
+Sale Datetime Confirm
```
Comment (elicoidal) [inline sale_datetime_confirm/README.rst]: Sales Confirmation Datetime field

File: sale_datetime_confirm/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+#                                                                       #
+# You should have received a copy of the                                #
+# GNU Affero General Public License                                     #
+# along with this program.  If not, see <http://www.gnu.org/licenses/>. #
+#                                                                       #
+#########################################################################
+
+
+{
+    'name': 'Sale Datetime Confirm',
```
Comment (elicoidal) [inline sale_datetime_confirm/__openerp__.py]: "Sales Confirmation Datetime field"

File: sale_datetime_confirm/model/sale.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+from openerp import fields, models, api
+from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT
+from datetime import datetime, timedelta
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    datetime_confirm = fields.Datetime(string='Confirm Date', copy=False)
```
Comment (elicoidal) [inline sale_datetime_confirm/model/sale.py]: 'Confirmation Date'

File: sale_datetime_confirm/view/sale_view.xml [hunk 7]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
```
Comment (elicoidal) [inline sale_datetime_confirm/view/sale_view.xml]: you can odoo instead of openerp

**Review by elicoidal — CHANGES_REQUESTED** [review]
Some minor details.
:+1:

**Comment by pedrobaeza** [comment]
Better to call it _sale_date_confirm_ instead, isn't it?

**Comment by lmignon** [comment]
@florian-dacosta The field is still there https://github.com/odoo/odoo/blob/9.0/addons/purchase/purchase.py#L110
But hidden for some obscure reasons https://github.com/odoo/odoo/blob/9.0/addons/purchase/purchase_view.xml#L270

**Comment by elicoidal** [comment]
@lmignon you are pointing to the purchase module: any reason?

**Comment by lmignon** [comment]
@elicoidal Oups sorry.... forget my comment. Thank you @elicoidal !

**Comment by florian-dacosta** [comment]
Thanks for the feedback, I have renamed the module and made the changes requested by @elicoidal

**Review by elicoidal — APPROVED** [review]

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM :+1: 
Please, squash commits 🙂

**Comment by florian-dacosta** [comment]
I'll close, since Odoo have put back the field in version 10

**Pull request closed without merging**